File: consecutive-characters/consecutive-characters.cpp
class Solution {
public:
    
    int maxPower(string s) {
        char curr_char = 'a';
        int ans = 0;
        int curr_max = 0;
        for(int i = 0 ; i < s.size() ; i++){
            if(s[i] == curr_char){
                curr_max++;
            }else{
                ans = max(ans,curr_max);
                curr_max = 1;
                curr_char = s[i];
            }
        }
         ans = max(ans,curr_max);
       return ans;
    }
};

File: consecutive-characters/consecutive-characters.go
func max(i,j int)int{
    if(i < j){
        return j
    }
    return i
}
func maxPower(s string) int {
    currChar := 'a'
    ans,currMax := 0,0
    length := len(s)
    for i := 0 ; i < length ; i++ {
        if s[i] == byte(currChar) {
            currMax++;
        }else{
            ans = max(ans,currMax)
            currMax = 1
            currChar = rune(s[i])
        }
    }
    ans = max(ans,currMax)
    return ans
}